### ISSUE TITLE
"string +=" on large strings is excessively slow in godot

### ISSUE THREAD

**Issue by RavinMaddHatter** [fbcosentino/godot-objexporter#3]
I will try and put together a pr for this, I need to clean it up a fair bit (and put back in the UV and texture export)

the use of the following pattern results in tons of memory allocation transactions that take longer as the model gets bigger:
`output += "small string"`

By switching from a bunch of concats to the following pattern i was able to reduce the run time by a factor of 4:
`output +="f %s %s %s" %  [xval, yval, zval]`

by using a temp string in the mesh writer instead of the full output string, i was able to increase the 60 triangle writes to 900 triangle writes without dropping frame rate on my PC. The pattern for that looks like this 

`var temp = ""
while i<indices_count:
     temp += "f  %s %s %s"%[x,y,z] #all of the code for the triangle here
     if (i%900)==0:
          output+=temp
          temp=""
          await get_tree().process_frame`

In the end without normals and texture (something i didn't care about for my application) i was able to drop export time from 300 seconds down to 20 seconds.

**Comment by onze** [comment]
If you could submit a test project we could benchmark a few ways to get this faster :)

**Comment by RavinMaddHatter** [comment]
Yeah I have it on my list of things to work out, I will put it in this
weekend
